src/epics/chatEpics.js

```diff
@@ -73,98 +67,34 @@ const suggestionEpic = (action$) =>
     .catch((err) => console.error(err));
 
 
-function createKeyPairObservable({ pincode = '' }) {
-  return Observable.create(function (observer) {
-    const urlHash = document.location.hash + pincode;
-    // 1. Get passphrase
-    const {
-      passphrase,
-      isNewPassphrase
-    } = getPassphrase(urlHash);
-
-    // 3. Get email based on passphrase
-    let email = getEmail(passphrase);
-    // 4. Decrypt to get key pair
-    miniLock.crypto.getKeyPair(passphrase, email, (keyPair) => {
-      miniLock.session.keys = keyPair;
-      miniLock.session.keyPairReady = true;
-      let mID = miniLock.crypto.getMiniLockID(keyPair.publicKey);
-      // 4. When we have keypair, login:
-
-      if (isNewPassphrase) {
-        document.location.hash = '#' + passphrase;
-      }
-
-      observer.next({ passphrase, keyPair, mID, isNewRoom: isNewPassphrase });
-      observer.complete();
-    });
-  });
-}
-
-function createDecryptMessageObservable({ message, mID, secretKey }) {
-  return Observable.create(function (observer) {
-    miniLock.crypto.decryptFile(message, mID, secretKey,
-      function (fileBlob, saveName, senderID) {
-        const reader = new FileReader();
-        reader.addEventListener("loadend", () => {
-          const authToken = reader.result;
-          observer.next(authToken);
-          observer.complete();
-        });
-
-        reader.readAsText(fileBlob);
-      });
+const initChatHandlerConnection = ({ authToken, secretKey, mID, isNewRoom }) => {
+  return chatHandler.initConnection({
+    authToken,
+    secretKey: secretKey,
+    mID
   });
-}
-
-function getAuthRequestSettings({ mID }) {
-  const settings = {
-    url: authUrl,
-    responseType: 'blob',
-    headers: {
-      'X-Minilock-Id': mID
-    },
-    method: 'GET'
-  };
-  return settings;
-}
+};
 
 const initConnectionEpic = (action$) =>
   action$.ofType(CHAT_INIT_CONNECTION)
```

Comment:
Step 6: This listens for this action to be triggered. We enter this flow with the action parameters. We pass those, first thing, to the initChatHandlerConnection method in here.